PostFreeSurfer/PostFreesurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,434 @@
+#!/bin/bash
+
+# # PreFreeSurferPipeline.sh
+#
+# ## Copyright Notice
+#
+# Copyright (C) 2022-2024 The Human Connectome Project
+#
+# * Washington University in St. Louis
+# * University of Minnesota
+# * Oxford University
+#
+# ## Author(s)
+#
+# * Mikhail Milchenko, Department of Radiology, Washington University in St. Louis
+# * Matthew F. Glasser, Department of Anatomy and Neurobiology, Washington University in St. Louis
+#
+# ## Product
+#
+# [Human Connectome Project][HCP] (HCP) Longitudinal Pipelines
+#
+# ## License
+#
+# See the [LICENSE](https://github.com/Washington-University/Pipelines/blob/master/LICENSE.md) file
+
+
+function create_zero_warp
+{
+    local target_image=$1 warp_file=$2
+
+    # Step 1: Extract dimensions from the target image
+    local dim1=$(fslval $target_image dim1)
+    local dim2=$(fslval $target_image dim2)
+    local dim3=$(fslval $target_image dim3)
+    local pixdim1=$(fslval $target_image pixdim1)
+    local pixdim2=$(fslval $target_image pixdim2)
+    local pixdim3=$(fslval $target_image pixdim3)
+
+    local temp=`mktemp volXXX`.nii.gz
+
+    # Step 2: Create a zero-filled image with the same dimensions
+    fslmaths $target_image -mul 0 $temp
+
+    # Step 3: Merge it into a 4D file with three volumes (all zeros)
+    fslmerge -t $warp_file $temp $temp $temp
+
+    # Optional: Set the voxel dimensions correctly if needed
+    fslcreatehd $dim1 $dim2 $dim3 3 $pixdim1 $pixdim2 $pixdim3 1 0 0 0 16 $warp_file
+}
+
+
+set -eu
+
+# Requirements for this script
+#  installed versions of: FSL
+#  environment: HCPPIPEDIR, FSLDIR
+
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/processingmodecheck.shlib" "$@" # Check processing mode requirements
+
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var FSLDIR
+log_Check_Env_Var HCPPIPEDIR_Global
+
+
+log_Msg "Platform Information Follows: "
+uname -a
+"$HCPPIPEDIR"/show_version
```

Comment:
With its current placement, this stuff will all show up before the usage info (and can cause the script to abort before displaying its usage).  Please move it down below "opts_ShowValues".

Reply:
ok